models/backbone/mobilenet.py

```diff
@@ -0,0 +1,218 @@
+    layers.append(
+        block(
+            inplanes,
+            planes,
+            stride=stride,
+            dilation=dilation,
+            use_gn=use_gn)
+    )
+    inplanes = planes
+    for i in range(1, blocks):
+        layers.append(
+            block(
+                inplanes,
+                planes,
+                stride=1,
+                dilation=dilation,
+                use_gn=use_gn)
+        )
+    return nn.Sequential(*layers)
+
+
+class MobileNet(nn.Module):
+    """
+    MobileNet backbone.
+    paper: https://arxiv.org/abs/1704.04861
+
+    Args:
+        width_multi (float): width_multi of mobilenet, from {0.25, 0.5, 0.75,
+            1.0}.
+        num_stages (int): mobilenet stages, normally 5.
+        strides (Sequence[int]): Strides of the first block of each stage.
+        dilations (Sequence[int]): Dilation of each stage.
+        out_indices (Sequence[int]): Output from which stages.
+        frozen_stages (int): Stages to be frozen (all param fixed). -1 means
+            not freezing any parameters.
+        use_gn (bool): Whether to use GN for normalization layers
+        bn_eval (bool): Whether to set BN layers to eval mode, namely, freeze
+            running stats (mean and var), only used when using BN.
+        bn_frozen (bool): Whether to freeze weight and bias of BN layers, only
+            used when using BN.
+    """
+
+    arch_settings = {
+        0.25: ((16, 32, 64, 128, 256), (1, 2, 2, 6, 2)),
+        0.5: ((32, 64, 128, 256, 512), (1, 2, 2, 6, 2)),
+        0.75: ((48, 96, 172, 384, 768), (1, 2, 2, 6, 2)),
+        1.0: ((64, 128, 256, 512, 1024), (1, 2, 2, 6, 2))
+    }
+
+    def __init__(self,
+                 width_multi=1.0,
+                 num_stages=5,
+                 strides=(1, 2, 2, 2, 2),
+                 dilations=(1, 1, 1, 1, 1),
+                 out_indices=(0, 1, 2, 3, 4),
+                 frozen_stages=-1,
+                 use_gn=False,
+                 bn_eval=True,
+                 bn_frozen=False):
+        super(MobileNet, self).__init__()
+        if width_multi not in self.arch_settings:
+            raise KeyError('invalid depth {} for mobilenet'.format(
+                width_multi))
+        assert 1 <= num_stages <= 5
+        assert len(strides) == len(dilations) == num_stages
+        assert max(out_indices) < num_stages
+
+        self.out_indices = out_indices
+        self.frozen_stages = frozen_stages
+        if not use_gn:
+            self.bn_eval = bn_eval
+            self.bn_frozen = bn_frozen
+        self.use_gn = use_gn
+
+        self.inplanes = round(32 * width_multi)
+        self.conv1 = conv3x3_group(3, round(32 * width_multi), stride=2)
+        self.norm1_name = 'bn1' if not use_gn else 'gn1'
+        self.add_module(self.norm1_name,
+                        norm_layer(round(32 * width_multi), use_gn))
+        self.relu = nn.ReLU(inplace=True)
+
+        stage_blocks, num_blocks = self.arch_settings[width_multi]
```

Comment:
I think here should be:
```
stage_outplanes, stage_blocks = self.arch_settings[width_multi]
```
And the use of `stage_blocks` in this implementation has minor bugs, you should fix it